core/src/main/java/org/apache/iceberg/FastAppend.java

```diff
@@ -128,7 +130,7 @@ protected void cleanUncommitted(Set<ManifestFile> committed) {
     }
   }
 
-  private ManifestFile writeManifest() throws IOException {
+  private ManifestFile writeManifest(long availSeqNum) throws IOException {
```

Comment:
Sequence numbers are not written into manifests during commit because the order of commits is not known until the commit succeeds. Retries require changing the sequence number.

To get around this, sequence numbers are written into the manifest list for each ManifestFile object, and are inherited when reading a manifest.

Reply:
Make sense.